core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -342,6 +343,15 @@ private[spark] class BlockManager(
             iter.close()
             false
         }
+      } catch {
+        case ex: KryoException
+          // We need to have clear error message to catch environmental problems easily.
+          // Further details: https://issues.apache.org/jira/browse/SPARK-37710
+          if ex.getMessage.toLowerCase(Locale.ROOT)
```

Comment:
Can we match on the cause of the KryoException to detect IOException? would that catch it more cleanly, or are only some IOExceptions of interest?

Reply:
Yes, we can check `IOException` at `Kryo` deserialization level for more generic error message. Updated.